docs/modules/general/pages/architecture-overview/mfe.adoc

```diff
@@ -1,37 +1,40 @@
-* **Remote Component display**: Any of Microfrontend's module, Remote Component or Shell Application in any period during platform runtime uses remote entry file to display a piece of user interface within a page.
+As previously stated, each Microfrontend is exposing a remote entry file, that can be accessed by any UI App. This remote entry file should contain all the data required to display the desired pieces of user interface. In the context of OneCX platform, this file should expose the following types of content:
 
-// TODO: Add link to remote components
-Module display is suited for whole UI modules, which consists of many pages and its own routing rules, whereas Remote Components are suited for smaller pieces of UI, which can be easily replacable and reusable within the platform. It is important to know that **Remote Components and Modules are both exposed via Microfrontend through a remote entry file**. More on the topic of Remote Components can be found here(TODO).
+* **Module**: Shell during workspace construction creates routing rules and associates a module from remote entry file with a specific route. Whenever user navigates to that route module is loaded using remote entry file to display desired content in the content pane of the page.
+* **Remote Component**: Component exposed via remote entry file is loaded onto a page on demand using Slot Component.
+
+Module type is suited for displaying whole UI modules containing many pages and its own routing rules, whereas Remote Component is suited for smaller pieces of UI, which can be easily replacable and reusable within the platform. It is important to know that **Remote Components and Modules are both exposed via a Microfrontend through a remote entry file**. More on the topic of Remote Components can be found xref:architecture-overview/remoteComponents.adoc[here].
 
 === Scaled agile
-The main reason of using approach with microfrontends is to allow teams to develop user interfaces in a scaled agile manner. It is best suited for large organisations that want to create user interfaces using many independent teams. In such cases, it is not possible, to develop an UI monolith application efficiently. With the Microfrontends approach, OneCX platform is delivering a solution to allow those teams to work on their own pieces of user interfaces, while preserving a common ground and allowing to easily integrate with eachother.
+The main reason of using approach with Microfrontends is to allow teams to develop user interfaces in a scaled agile manner. It is best suited for large organisations that want to create user interfaces using many independent teams. In such cases, it is not possible, to develop an UI monolith application efficiently. With the Microfrontends approach, OneCX platform is delivering a solution to allow those teams to work on their own pieces of user interfaces, while preserving a common ground and allowing to easily integrate with eachother.
```

Comment:
The main reason of using the Microfrontends approach is to allow teams to develop user interfaces in a scaled agile manner.

... with each other.